src/account-info/AccountDetailsScreen.js

```diff
@@ -74,7 +76,10 @@ export default function AccountDetailsScreen(props: Props): Node {
 
   return (
     <Screen title={title}>
-      <AccountDetails user={user} />
+      <AccountDetails
+        user={user}
+        displayUserEmail={emailAddressVisibility === EmailAddressVisibility.Everyone}
```

Comment:
Relatedly: this condition isn't sufficient for determining whether the `email` property is a fake placeholder which (as mentioned on the issue description at #5400) shouldn't be shown.

In particular, if `realm_email_address_visibility` has a value like Admins (which it does on chat.zulip.org, for example), then org admins have the ability to see people's actual email addresses, and so we should have mobile work for them in that case too.

The logic here is a little bit tricky, because depending on settings the correct email address to show may appear on either `user.email` or another property, `user.delivery_email`. We also [have some plans](https://chat.zulip.org/#narrow/stream/378-api-design/topic/email.20address.20visibility/near/1296133) to change how those work in the future, in order to support users individually choosing whether their email addresses are visible. Having just gone and refreshed myself on this part of the API and those plans I think the right logic to accomplish what we want here is:
* If there is a `delivery_email` property and its value is `null`, then we don't have access to the real email address; don't try to show one.
  * This case doesn't happen on current servers, but is planned for the future; it's easy to go ahead and handle, so might as well.
  * Currently we declare the `delivery_email` property in `src/api/modelTypes.js` as `+delivery_email?: string`. In order to let the type-checker know that null is a possibility as well as strings, let's change `string` there to `string | null`. A comment should note this is a projected future API, with a link to the same `#api design` chat thread I linked above.
* Otherwise, if there is a `delivery_email` property, then it must be the actual email address; use that.
* Otherwise, if `realm_email_address_visibility` is Everyone, then the `email` property is the actual email address; use that.
  * On future servers, we expect this case will never happen: we'll always include a `delivery_email` when you have access, including when the visibility is Everyone.
* Otherwise, we don't have access to the actual email address; don't try to show one.

That logic should all get encapsulated in a function of its own, returning `string | null` -- the email address to show if there is one, else null. A good home for it would be `src/users/userSelectors.js`, with a name like `getDisplayEmailForUser`; it can take a `RealmState` and `UserOrBot` as arguments, similar to the existing `getCustomProfileFieldsForUser`.

That function should also get some unit tests. See the same existing function for comparison (though I think this won't need as many test cases as that existing function does.)

Reply:
Here is a draft `getDisplayEmailForUser`.  

One thing I noticed is that the return type for flow needed to have `string` be a `?string` because delivery_email is annotated as a `?string | null`.  And it makes perfect sense that flow isn't going to execute the code and be smart enough to know we always return string or null but never undefined and it's flagging that.  Is it possible to tell Flow we really know what we're doing and want to not have string be a maybe type?

```
export function getDisplayEmailForUser(realm: RealmState, user: UserOrBot): ?string | null {
  const { emailAddressVisibility } = realm;

  if ('delivery_email' in user) {
    const { delivery_email } = user;
    if (delivery_email === null) {
      return null;
    } else {
      return delivery_email;
    }
  } else if (emailAddressVisibility === EmailAddressVisibility.Everyone) {
    // On future servers, we expect this case will never happen: we'll always include
    // a delivery_email when you have access, including when the visibility is Everyone
    // https://github.com/zulip/zulip-mobile/pull/5515#discussion_r997731727

    return user.email;
  } else {
    return null;
  }
}
```

Reply:
I tested this on The Recurse Center server (email address visibility = everyone) and Zulips server (email address visibility = admins) and it seems to work properly.

Reply:
Let's continue this in a chat thread, for more convenient discussion:
https://chat.zulip.org/#narrow/stream/243-mobile-team/topic/finding.20display.20email/near/1453823